### DISCUSSION TITLE
Possible Extend the Section component (in Oqtane) with "Visible" parameter

### DISCUSSION THREAD

**Discussion by beolafsen** [oqtane/oqtane.framework#3520] [Ideas]
Hi!
Would it be posible to extend the "Section" component with a "IsVisible" parameter?
I can make my own Section but if it is usefull for others it is maybe possible to add it into existing component?

[Parameter]
public bool IsVisible { get; set; } = true;

**Comment by sbwalker** [comment]
@beolafsen I assume you want to control the visibility of a section within your module component based on some sort of custom logic or boolean value. It is obviously already possible to hide/show sections using the following syntax: 

```
@if (IsVisible)
{
    <Section Name="Name">
        ...
    </Section>
}
```

and you are requesting to be able to do something like this instead:

```
<Section Name="Name" IsVisible="@IsVisible">
    ...
</Section>
```

Are you simply trying to avoid writing conditional logic in your module components? I am trying to understand the use case.

**Comment by beolafsen** [comment]
Hi!
Yes – you're right. My logic today  is how you describe :

@if (IsVisible)
{
    <Section Name="Name">
        ...
    </Section>

}
.
I have about 10-15 of them in succession and I guess I like the syntac
better when the IF logic is put in the component.
Not a big problem, but I like it better.
I have already made a copy of your Section component - so it's not a big
deal.
Nearly all my components have this IsVisible parameter - so I guess I like
it better. :-)

Best regards
Bjørn-Even Olafsen


tor. 30. nov. 2023 kl. 21:23 skrev Shaun Walker ***@***.***>:

**Comment by sbwalker** [comment]
@beolafsen I can see the benefit of easier to read/maintain code... please go ahead and submit the PR

**Comment by beolafsen** [comment]
Great - I have done that.
It's my first contribution so give me a hint if I have done something wrong.

Best regards
Bjørn-Even Olafsen


tor. 30. nov. 2023 kl. 23:20 skrev Shaun Walker ***@***.***>:
